src/server/JasmineGraphInstanceService.cpp

```diff
@@ -741,185 +767,127 @@ void *instanceservicesession(void *dummyPt) {
-            centralDB = graphDBMapCentralStores[graphID + "_centralstore_" + partitionID];
+            write(connFd, JasmineGraphInstanceProtocol::OK.c_str(), JasmineGraphInstanceProtocol::OK.size());
+            instance_logger.log("Sent : " + JasmineGraphInstanceProtocol::OK, "info");
+            bzero(data, INSTANCE_DATA_LENGTH);
+            read(connFd, data, INSTANCE_DATA_LENGTH);
+            string graphID = (data);
+            graphID = utils.trim_copy(graphID, " \f\n\r\t\v");
+            instance_logger.log("Received Graph ID: " + graphID, "info");
 
-            map<long, long> degreeDistribution = graphDB.getInDegreeDistributionHashMap();
-            std::map<long, long>::iterator its;
+            write(connFd, JasmineGraphInstanceProtocol::OK.c_str(), JasmineGraphInstanceProtocol::OK.size());
+            instance_logger.log("Sent : " + JasmineGraphInstanceProtocol::OK, "info");
 
-            map<long, long> degreeDistributionCentral = centralDB.getInDegreeDistributionHashMap();
-            std::map<long, long>::iterator itcentral;
+            bzero(data, INSTANCE_DATA_LENGTH);
+            read(connFd, data, INSTANCE_DATA_LENGTH);
+            string partitionID = (data);
+            partitionID = utils.trim_copy(partitionID, " \f\n\r\t\v");
+            instance_logger.log("Received Partition ID: " + partitionID, "info");
 
-            for (its = degreeDistributionCentral.begin(); its != degreeDistributionCentral.end(); ++its) {
-                bool centralNodeFound = false;
-                for (itcentral = degreeDistribution.begin(); itcentral != degreeDistribution.end(); ++itcentral) {
-                    if ((its->first) == (itcentral->first)) {
-                        degreeDistribution[its->first] = (its->second) + (itcentral->second);
-                        centralNodeFound = true;
-                    }
-                }
-                if (!centralNodeFound) {
-                    degreeDistribution.insert(std::make_pair(its->first, its->second));
-                }
-            }
+            write(connFd, JasmineGraphInstanceProtocol::OK.c_str(), JasmineGraphInstanceProtocol::OK.size());
+            instance_logger.log("Sent : " + JasmineGraphInstanceProtocol::OK, "info");
 
-            // Invoke other workers to calculate their own our degree distributions
-            // TODO(kasundharmadasa:  invoke other workers asynchronously)
-            for (vector<string>::iterator workerIt = workerSockets.begin(); workerIt != workerSockets.end();
-                 ++workerIt) {
-                std::vector<string> workerSocketPair;
-                stringstream wl(*workerIt);
-                string intermediate;
-                while (getline(wl, intermediate, ':')) {
-                    workerSocketPair.push_back(intermediate);
-                }
+            bzero(data, INSTANCE_DATA_LENGTH);
+            read(connFd, data, INSTANCE_DATA_LENGTH);
+            string workerList = (data);
```

Comment:
Can't we just change this line as follows,

`string workerList = data;`

Reply:
resolved

Reply:
Ok